hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/connector/map/MetadataCompactResolver.java

```diff
@@ -70,12 +75,17 @@ private MetadataCompactResolver() {
             throw QueryException.error("Unable to resolve table metadata. Missing ['typeName'] option");
         }
 
-        return extractFields(userFields, isKey).entrySet().stream()
+        Map<QueryPath, MappingField> fields = extractFields(userFields, isKey);
+        return fields.entrySet().stream()
                 .map(entry -> {
                     QueryPath path = entry.getKey();
                     if (path.getPath() == null) {
                         throw QueryException.error("Cannot use the '" + path + "' field with Compact serialization");
                     }
+                    QueryDataType type = entry.getValue().type();
```

Comment:
What about `key`? Can it be of type OBJECT?

Reply:
This path is checking both `key` and `value`. Check the callers of the method, and `usage isKey` parameter of the `resolveAndValidateFields`